github/resource_github_issue_label.go

```diff
@@ -75,10 +74,8 @@ func resourceGithubIssueLabelCreateOrUpdate(d *schema.ResourceData, meta interfa
 		Name:  github.String(name),
 		Color: github.String(color),
 	}
-	ctx := context.Background()
-	if !d.IsNewResource() {
-		ctx = context.WithValue(ctx, ctxId, d.Id())
-	}
+
+	ctx := prepareResourceContext(d)
```

Comment:
This replacement seems to have caused the failing acceptance test.  Reverting this hunk got tests passing for me but I cannot spot what causes the failure:

```

	// func prepareResourceContext(d *schema.ResourceData) context.Context {
	// 	ctx := context.WithValue(context.Background(), ctxId, d.Id())
	// 	if etag, set := d.GetOk(ctxEtag); set {
	// 		ctx = context.WithValue(ctx, ctxEtag, etag.(string))
	// 	}
	//
	// 	return ctx
	// }

	// Tests fail
	ctx := prepareResourceContext(d)

	// Tests pass on previous logic
	ctx = context.Background()
	if !d.IsNewResource() {
		ctx = context.WithValue(ctx, ctxId, d.Id())
	}

```
```
--- FAIL: TestAccGithubIssueLabel_description (9.71s)
    testing.go:569: Step 1 error: errors during apply:

        Error: GET https://api.github.com/repos/github-terraform-staging/tf-acc-test-branch-issue-label-desc-y2aib/labels/foo: 304  []

          on /var/folders/t1/p9cyhv6s2wg4g8s_x6zkpz_m0000gn/T/tf-test562137662/main.tf line 6:
          (source code not available)


FAIL
FAIL	github.com/terraform-providers/terraform-provider-github/github	10.026s
```
> 2020/02/03 18:19:07 [DEBUG] github_issue_label.test: apply errored, but we're indicating that via the Error pointer rather than returning it: GET https://api.github.com/repos/github-terraform-staging/tf-acc-test-branch-issue-label-desc-un80a/labels/foo: 304  []